authorize/evaluator/config.go

```diff
@@ -17,6 +19,7 @@ type evaluatorConfig struct {
 	JWTClaimsHeaders                                  config.JWTClaimHeaders
 	JWTGroupsFilter                                   config.JWTGroupsFilter
 	DefaultJWTIssuerFormat                            config.JWTIssuerFormat
+	MCPAccessTokenProvider                            func(string, time.Time) (string, error) `hash:"-"`
```

Comment:
in case of MCP server upstream, we would use this function to set the authorization header with the access token of an external oauth2 provider such as github/notion/etc